server/zanata-war/src/test/java/org/zanata/adapter/PropertiesLatinOneAdapterTest.java

```diff
@@ -22,44 +22,104 @@
         assertThat(resource.getTextFlows()).hasSize(3);
         assertThat(resource.getTextFlows().get(0).getId()).isEqualTo(
                 "line1");
         assertThat(resource.getTextFlows().get(0).getContents()).isEqualTo(
                 ImmutableList.of("Line One"));
     }
 
+    /*
+     * Properties files change path, not name
+     */
+    @Test
+    public void testGeneratedFilename() throws Exception {
+        HDocument document = new HDocument("/test/basicprop.properties",
+                "basicprop.properties", "test/", ContentType.PO,
+                new HLocale(new org.zanata.common.LocaleId("en")));
+        HRawDocument hRawDocument = new HRawDocument();
+        hRawDocument.setType(DocumentType.PROPERTIES);
+        document.setRawDocument(hRawDocument);
+
+        assertThat(adapter.generateTranslationFilename(document, "fr"))
+                .isEqualTo("basicprop.properties");
+    }
+
+    @Test
+    public void testTranslatedPropertiesDocument() {
+        TranslationsResource tResource = new TranslationsResource();
+        addTranslation(tResource, "line1", "Founde metalkcta", ContentState.Approved);
+        addTranslation(tResource, "line2", "Tbade metalkcta", ContentState.Translated);
+        addTranslation(tResource, "line3", "Kbade metalkcta", ContentState.NeedReview);
+
+        Resource resource = parseTestFile("test-properties-latin1.properties");
+        File originalFile = new File(resourcePath.concat("test-properties-latin1.properties"));
+        OutputStream outputStream = new ByteArrayOutputStream();
+
+        adapter.writeTranslatedFile(outputStream,
+                originalFile.toURI(),
+                resource,
+                tResource,
+                "ru",
+                Optional.absent());
+
+        assertThat(outputStream.toString()).isEqualTo(
+                "line1=Founde metalkcta\nline2=Tbade metalkcta\nline3=\n");
+    }
+
+    @Test
+    public void testLatin1encoding() throws Exception {
+        File latin1EncodedFile = createTempFile(StandardCharsets.ISO_8859_1);
+        Resource resource =
+                adapter.parseDocumentFile(latin1EncodedFile.toURI(), LocaleId.EN,
+                        Optional.absent());
+        assertThat(resource.getTextFlows().get(0).getId()).isEqualTo(
+                "line1");
+        assertThat(resource.getTextFlows().get(0).getContents())
+                .containsExactly("Line One");
+    }
+
+    @Test
+    public void testUTFOnLatin1encoding() throws Exception {
+        File latin1EncodedFile = createTempFile(StandardCharsets.UTF_8);
+        Resource resource =
+                adapter.parseDocumentFile(latin1EncodedFile.toURI(), LocaleId.EN,
+                        Optional.absent());
+        assertThat(resource.getTextFlows().get(0).getId()).isEqualTo(
+                "line1");
+        assertThat(resource.getTextFlows().get(0).getContents())
+                .containsExactly("Â¥Line One");
```

Comment:
I don't think it's a good idea to test that a file gets corrupted with a particular corruption. Better to test that we can read and write the character "¾" correctly via Latin 1, for instance.